### DISCUSSION TITLE
Unexpected LONGTEXT type

### DISCUSSION THREAD

**Discussion by wu-clan** [sqlalchemy/alembic#1584] [Usage Questions] [answered]
**Describe the bug**

**Expected behavior**

**To Reproduce**
Please try to provide a [Minimal, Complete, and Verifiable](http://stackoverflow.com/help/mcve) example, with the migration script and/or the SQLAlchemy tables or models involved.
See also [Reporting Bugs](https://www.sqlalchemy.org/participate.html#bugs) on the website.

```python
class Api(Base):
    """Sys api"""

    __tablename__ = 'sys_api'

    id: Mapped[int] = mapped_column(primary_key=True, index=True, autoincrement=True)
    remark: Mapped[str | None] = mapped_column(LONGTEXT().with_variant(TEXT, 'postgresql'))
```

**Error**

```python
# In versions py
sa.Column('remark', mysql.LONGTEXT(), nullable=True)
```

```
sqlalchemy.exc.CompileError: (in table 'sys_api', column 'remark'): Compiler <sqlalchemy.dialects.postgresql.base.PGTypeCompiler object at 0x000001D463CA40D0> can't render element of type LONGTEXT
```

**Versions.**
 - OS: Windows11
 - Python: 3.10
 - Alembic: 1.14.0
 - SQLAlchemy: 2.0.30
 - Database: postgresql
 - DBAPI: asyncpg

**Additional context**

postgresql+asyncpg://postgres:123456@127.0.0.1:5432/fba

**Have a nice day!**

**Comment by zzzeek** [comment, marked as answer]
alembic seems to be skipping the reverse variant at the moment.  please set up the variant with generic type first as this is what alembic expects right now:

     TEXT().with_variant(LONGTEXT(), "mysql")

this will be fixed by #1585 will be in alembic 1.14.1
